core/quantity/test/Parsing.test.ts

```diff
@@ -172,30 +171,64 @@ describe("Parsing tests:", () => {
 
     const testData = [
       { value: "12,345.345", quantity: { magnitude: 12345.345, unitName: "Units.FT" } },
+      { value: "3.6252e3 Miles", quantity: { magnitude: 19141056, unitName: "Units.FT" } },
+      { value: "3.6252e-3 Miles", quantity: { magnitude: 19.141056, unitName: "Units.FT" } },
+      { value: "3.6252e+3 Miles", quantity: { magnitude: 19141056, unitName: "Units.FT" } },
+      { value: "-1 1/2FT", quantity: { magnitude: -1.5, unitName: "Units.FT" } },
+      { value: "-2FT 6IN", quantity: { magnitude: -2.5, unitName: "Units.FT" } },
+      { value: "1 1/2 FT", quantity: { magnitude: 1.5, unitName: "Units.FT" } },
+      { value: "2FT 6IN", quantity: { magnitude: 2.5, unitName: "Units.FT" } },
+      { value: `2'-6"`, quantity: { magnitude: 2.5, unitName: "Units.FT" } },
+      { value: "-3IN", quantity: { magnitude: -0.25, unitName: "Units.FT" } },
+    ];
+
+    const unitsProvider = new TestUnitsProvider();
+    const format = new Format("test");
+    await format.fromJSON(unitsProvider, formatData).catch(() => { });
+    assert.isTrue(format.hasUnits);
+
+    for (const testEntry of testData) {
+      const quantityProps = await Parser.parseIntoQuantity(testEntry.value, format, unitsProvider);
+      // console.log (`quantityProps=${JSON.stringify(quantityProps)}`);
+      expect(Math.fround(quantityProps.magnitude)).to.eql(Math.fround(testEntry.quantity!.magnitude));
+      expect(quantityProps.unit.name).to.eql(testEntry.quantity!.unitName);
+    }
+  });
+
+  it("Parse into Length Quantities", async () => {
+    const formatData = {
+      formatTraits: ["keepSingleZero", "applyRounding", "showUnitLabel"],
+      precision: 4,
+      type: "Decimal",
+      uomSeparator: "",
+    };
+
+    const testData = [
+      { value: "12,345.345 FT", quantity: { magnitude: 12345.345, unitName: "Units.FT" } },
       { value: "3.6252e3 Miles", quantity: { magnitude: 3625.2, unitName: "Units.MILE" } },
       { value: "3.6252e-3 Miles", quantity: { magnitude: 0.0036252, unitName: "Units.MILE" } },
       { value: "3.6252e+3 Miles", quantity: { magnitude: 3625.2, unitName: "Units.MILE" } },
       { value: "-1 1/2FT", quantity: { magnitude: -1.5, unitName: "Units.FT" } },
       { value: "-2FT 6IN", quantity: { magnitude: -2.5, unitName: "Units.FT" } },
       { value: "1 1/2 FT", quantity: { magnitude: 1.5, unitName: "Units.FT" } },
       { value: "2FT 6IN", quantity: { magnitude: 2.5, unitName: "Units.FT" } },
-      { value: `2'-6"`, quantity: { magnitude: 2.5, unitName: "Units.FT" } },
+  /*    { value: `2'-6"`, quantity: { magnitude: 2.5, unitName: "Units.FT" } }, ambiguous case since ' can be Unit.FT or Units.ARC_MINUTE would require a Format unit to be defined. */
```

Comment:
Should this be a case where a preferred Unit is used?  I didn't quite follow on the initial review where the preferred unit comes from.

Reply:
No, not in this case because there is no unit defined in the Format... that means there is no unit to preference. In the test above this one the Format was a composite with unit[0] set to Units.FT so it had a preference.

Reply:
Ah okay, got it.

Reply:
I should also note that in imodeljs frontend and ui packages we use the other method of parsing where we pass in the persistence unit and in that case the persistence unit would serve as a fallback unit if the Format did not define one.